compare_attempts.py

```diff
@@ -0,0 +1,120 @@
+import csv
+import os
+import statistics
+import sys
+from typing import List
+
+import pandas as pd
+from dotenv import load_dotenv
+
+import utils
+from project_types.custom_types import Approach, Task, bcolors
+
+
+def compare(data_folder_path: str, samples_per_task: int):
+    header = [
+        "Filename",
+        "Total Tasks",
+        "Total Samples",
+        "Vulnerable Samples",
+        "Expected CWE Samples",
+        "Sample Vulnerable Percentages",
+        "Min Vulnerable Percentage",
+        "Median Vulnerable Percentage",
+        "Average Vulnerable Percentage",
+        "Max Vulnerable Percentage",
+        "Min Expected CWE Percentage",
+        "Median Expected CWE Percentage",
+        "Average Expected CWE Percentage",
+        "Max Expected CWE Percentage",
+    ]
+
+    matrix = []
+
+    for file in os.listdir(data_folder_path):
+        data_file_path = os.path.join(data_folder_path, file)
+        # checking if it is a file
+        if os.path.isfile(data_file_path):
+            print(f"Analyzing: {data_file_path}")
+            approach = utils.read_approaches_file(data_file_path)
+            if (
+                not approach.vulnerable_percentage is None
```

Comment:
use utils.validate_task_integrity and utils.validate_sample_integrity to make sure all files are valid before proceeding

Reply:
fixed